bin/generate_inputs.py

```diff
@@ -0,0 +1,221 @@
+            return json_data
+        else:
+            print("Could not retrieve xml for accession", assembly)
+            print(r.text)
+            return None
+
+
+def generate_software_input(erz_acc):
+    handler_request = handler.get_assembly(erz_acc)
+    assembly_software = handler_request["assembly_software"]
+    if not assembly_software:
+        json_analysis = load_xml(erz_acc)
+        assembly_software = json_analysis["ANALYSIS_SET"]["ANALYSIS"]["ANALYSIS_TYPE"]["SEQUENCE_ASSEMBLY"][
+            "PROGRAM"]
+        assembly_software = assembly_software.replace(" ", "_" if "v" in assembly_software else "_v")
+        if '_v' not in assembly_software:
+            assembly_software = assembly_software.replace('v', '_v')
+    return assembly_software
+
+
+def write_samplesheet_line(assembly, assembly_path, run, run_path, samplesheet):
+    with open(samplesheet, 'a') as file_out:
+        if assembly_path:
+            # TODO: return s3 when nextflow will work with -resume
+            transformed_assembly, _ = transform_paths(assembly_path)
+        else:
+            print(f'no assembly path {assembly}')
+        chosen_run_path = []
+        if run_path:
+            se, pe_forward, pe_reversed = False, False, False
+            if len(run_path) >= 2:
+                for raw_file in run_path:
+                    if len(raw_file.split('_1')) >= 2:
+                        pe_forward = raw_file
+                    elif len(raw_file.split('_2')) >= 2:
+                        pe_reversed = raw_file
+                    else:
+                        se = raw_file
+                if se and pe_forward and pe_reversed:
+                    print(f"More than 3 raw files detected for {run}, choosing PE reads")
+                    chosen_run_path = [pe_forward, pe_reversed]
+                elif pe_forward and pe_reversed:
+                    chosen_run_path = [pe_forward, pe_reversed]
+                else:
+                    chosen_run_path = [se]
+            else:
+                chosen_run_path = [run_path[0]]
+        else:
+            print(f'no runs path {assembly}')
+
+        transformed_runs = []
+        for item in chosen_run_path:
+            # TODO: return when s3 will work with -resume
+            transformed_path, _ = transform_paths(item)
+            transformed_runs.append(transformed_path)
+
+        line = f"{run},"
+        if len(transformed_runs) == 2:
+            line += f"{transformed_runs[0]},{transformed_runs[1]},"
+        elif len(transformed_runs) == 1:
+            line += f"{transformed_runs[0]},,"
+        else:
+            print("wrong length of runs path")
+        assembler = generate_software_input(assembly)
+        line += f"{assembly},{transformed_assembly},{assembler}"
+        file_out.write(line + '\n')
+
+
+def generate_lists(raw_reads_study, assembly_study, outdir, input_scientific_name, input_env_biome, keep_metat, samplesheet_path):
```

Comment:
As per my other comment.. I think it would improve the readability to split this into 2 methods.. one to source/massage the data and another one to write it.

Easiest to unit test too